AsyncNats/NatsConnection.cs

```diff
@@ -482,63 +544,90 @@ private async IAsyncEnumerable<T> InternalSubscribe<T>(string subject, string? q
                 try
                 {
                     await SendUnsubscribe(subscription);
-                    _subscriptions = _subscriptions.Where(s =>s != subscription).ToArray();
+                    _subscriptions.TryRemove(subscription.SubscriptionId, out _);
                 }
                 finally
                 {
                     _subscriptionsLock.Release();
                 }
             }
         }
+        private T DeserializeWrapper<T>(Func<NatsMsg, T> deserialize, NatsMsg message)
```

Comment:
What's the added benefit of this function? Making InternalSubscribe slightly less complex?